### ISSUE TITLE
play after x songs doesnt work as expected

### ISSUE THREAD

**Issue by nunotfc** [AzuraCast/AzuraCast#4796]
### Installation Method

Docker Installation

### AzuraCast Release Channel

Rolling Release Channel

### Current AzuraCast Version

266305e (2021-11-18 18:57) • Docker • PHP 8.0

### What happened?

dont know how this is appening, but i have quewe length set to 3. i have a playlist of jingles thatr plays every 2 songs. Actually it is playing 2 jingles after 1 song or a jingle after 2 songs. sometimes is correct. i have a 80s playlist that plays one song each 30 songs, also not working. dont know if it is related to jingle mode bug, or... if i set the quewe length to 1, all worked before. this is not a liquidsoap2 bug, i had it before but i solved it changing quewe length to 1. now it doesnt work also.

**Comment by YosefAdPro** [comment]
A known issue and there are some similar issues.
https://github.com/AzuraCast/AzuraCast/issues/4777
https://github.com/AzuraCast/AzuraCast/issues/4765

**Comment by SpeedyCharly** [comment]
I have reported that same issue a while back myself.  Seems to me the problem is that the file count is incremented by jingles as well as by songs which throws the whole thing out of whack. I have worked with clockwheels on apps such as SamBroadcaster where such problems don't arise because the jingle itself doesn't increment the count.  If developers see fit to change this so that the behavior becomes more predictable, please also make it so that jingles (short media clips) are excluded from the crossfade routine as well...

**Comment by nunotfc** [comment]
I AGREE WITH ALL.
the crossfade in jingles is so bad.

Às 16:36 de 23/11/2021, Speedy Charly escreveu:

**Comment by Gregory-Hepicloud** [comment]
Same problem setup playlist to jingle mode select play x song set to 2 and some time is song jingle song and not song song jingle

**Comment by SlvrEagle23** [comment]
Let's try to make sure every single GitHub Issue corresponds to one bug that we can work on as a team. The other commentary about crossfades and jingles not working correctly should be another issue (though it likely is already).

In our latest Rolling Release version, I identified a potential source of problems with the "once per x songs" playlist type and was able to update it within the last 24 hours; if your installations are experiencing problems with this playlist type, it may be worthwhile to update and see if things are improved. In my local testing, it returns everything to functioning exactly as expected.

**Comment by SpeedyCharly** [comment]
Sorry; my bad for introducing the 'crossfade on jingles' issue here.  Very real issue, but doesn't belong here.  

On the 'once per X issue'; I've just updated and, after allowing the stream to stabilize, I'm happy to report that the song rotation is now running as expected within the parameters I've set. But my 'Jingle' playlists are still disabled and I have yet to see what will happen when they're added back into the mix.
